### PR TITLE
MAINT: add test for 12-byte alignment

### PR DESCRIPTION
**Pull request by ahaldane**
Followup to #12611 to add a test for a 12-byte, 12-byte-aligned array.

I checked it crashes on x64 before #12611.

### CODE REVIEW

File: numpy/core/tests/test_multiarray.py [hunk 1]
```diff
@@ -8000,10 +8000,10 @@ def check(self, shape, dtype, order, align):
             raise ValueError()
 
     def test_various_alignments(self):
-        for align in [1, 2, 3, 4, 8, 16, 32, 64, None]:
+        for align in [1, 2, 3, 4, 8, 12, 16, 32, 64, None]:
             for n in [0, 1, 3, 11]:
                 for order in ["C", "F", None]:
-                    for dtype in np.typecodes["All"]:
+                    for dtype in list(np.typecodes["All"]) + ['i4,i4,i4']:
```
Comment (charris) [inline numpy/core/tests/test_multiarray.py]: Why is this needed?

Comment (ahaldane) [reply numpy/core/tests/test_multiarray.py]: I thought it would be good to throw in a 12-byte-uint-aligned type, and `i4,i4,i4` is 12 bytes.

**Comment by charris** [comment]
Looks like we have an odd invalid value `RuntimeWarning` slipping through from somewhere in Windows 32 bit, Python 3.7. Same thing turned up in a different place in PR previous to this one. I think it is irrelevant to this PR, but it would be nice to know where it comes from.

**Comment by charris** [comment]
Will rebuild azure test.

**Comment by charris** [comment]
To summarize the odd recurring test failures, the common parts seem to be

- `cgemv`
- Python 3.7
- OpenBLAS (3.4+)

I'm not sure what role the Python version might play in this.

**Comment by eric-wieser** [comment]
`np.dtype('i4,i4,i4').alignment == 4`. Is that relevant here?

**Comment by charris** [comment]
Thanks Allan.

**Comment by charris** [comment]
@ahaldane The tests passed here, but the [merge to master](https://travis-ci.org/numpy/numpy/builds/472723363) is failing, same for the backport. I assume there has been some sort of merge mixup with the previous merges. Could you take a look?

**Comment by charris** [comment]
I'm guess there are some extra flags being set for the merged version, but darned if I know what.

**Comment by ahaldane** [comment]
Strange. It could be a sporadic error I suppose, that only appears sometimes. Eg, it only happens if malloc gives back an unexpectedly aligned pointer. But I though we were forcing the alignment with `_aligned_zeros` ...

**Comment by ahaldane** [comment]
OK, I just ran tests with USE_DEBUG and I seem to be able to reproduce the crash.

**Comment by charris** [comment]
What is also odd is that I just tested a PR against the 1.16.x branch that now has the troublesome test in it, and it [passed](https://travis-ci.org/numpy/numpy/builds/472805521?utm_source=github_status&utm_medium=notification). So there is something different about the branch merges as opposed to the PRs.

**Comment by ahaldane** [comment]
Ok, it only turns up half the time because of malloc alignment variation, combined with the fact that the test was requesting 12 byte alignment and testing 16 byte alignment. But when you request 12 byte alignment, some of the time that is also 16 byte aligned, eg for pointer multiples of 48.
